### PR TITLE
[SPARK-38670][SS] Add offset commit time to streaming query listener

### PR DESCRIPTION
**Pull request by jerrypeng**
### What changes were proposed in this pull request?
Add a metric called "commitOffsets" to the StreamingQueryListener

### Why are the changes needed?
A good portion of the batch duration is committing offsets at the end of the micro-batch.  The timing for this operation is missing from the durationMs metrics.  Lets add this metric to have a more complete picture of where the time is going during the processing of a micro-batch

### Does this PR introduce _any_ user-facing change?
Yes, an event returned from the StreamingQueryListener will now contain an additional metric:

```
  "durationMs" : {
    "addBatch" : 39,
    "getBatch" : 0,
    "latestOffset" : 0,
    "queryPlanning" : 9,
    "triggerExecution" : 228,
    "walCommit" : 82,
    "commitOffsets": 81 // <---- added this metric
  },
```

### How was this patch tested?
add a test

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingQuerySuite.scala [hunk 1]
```diff
@@ -316,6 +316,7 @@ class StreamingQuerySuite extends StreamTest with BeforeAndAfter with Logging wi
         assert(query.recentProgress.last.eq(query.lastProgress))
 
         val progress = query.lastProgress
+
```
Comment (HeartSaVioR) [inline sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingQuerySuite.scala:319]: nit: unnecessary change

**Review by HeartSaVioR — COMMENTED** [review]
1. 
Please add `[SS]` after `[SPARK-38670]`.

2.
We have a stack graph with regarding to the duration in the micro batch in SS UI - I'm not sure the change is reflected on the graph.

Could you please run a e2e manually and make sure the change is also reflected in the SS UI page as well? Please attach the screenshot and the step you've performed on manual test in the section of "How was this patch tested?".

3.
This PR does introduce the user-facing change since we add the new field for durationMs in streaming listener. Could you please update the section "Does this PR introduce any user-facing change?" to describe the change?

Thanks in advance!

File: sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingQuerySuite.scala (hunk 1 above)
Comment (jerrypeng) [reply sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingQuerySuite.scala:319]: will revert

**Comment by jerrypeng** [comment]
Screen Shot 2022-03-28 at 3 58 17 PM

An image from the UI, the new metric shows up correctly.

**Comment by jerrypeng** [comment]
@HeartSaVioR thanks for the review.  Can you please take another look?

**Review by HeartSaVioR — APPROVED** [review]
+1

**Comment by HeartSaVioR** [comment]
Thanks! Merging to master.

**Comment by HeartSaVioR** [comment]
Thanks @jerrypeng for your contribution! I merged this to master.
